lib/data/external/GCP/GcpApis/completeMPU.js

```diff
@@ -0,0 +1,232 @@
+        logger.trace('completeMultipartUpload: overflow compose');
+        const parts = createMpuList(params, 'overflow', numParts);
+        const partList = parts.map(item => (
+            { PartName: item.PartName }));
+        if (partList.length < 2) {
+            logger.trace(
+                'fewer than 2 parts in overflow, skip to copy phase');
+            return callback(null, partList[0].PartName);
+        }
+        const composeParams = {
+            Bucket: params.Overflow,
+            Key: createMpuKey(params.Key, params.UploadId, 'final'),
+            MultipartUpload: { Parts: partList },
+        };
+        return _retryCompose.call(this, composeParams, 0, err => {
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, null);
+        });
+    };
+
+    /*
+     * Create MPU Aggregate ETag
+     */
+    const _generateMpuResult = (res, callback) => {
+        const concatETag = partList.reduce((prev, curr) =>
+            prev + curr.ETag.substring(1, curr.ETag.length - 1), '');
+        const aggregateETag = createAggregateETag(concatETag, partList);
+        return callback(null, res, aggregateETag);
+    };
+
+    const _copyToMain = (res, aggregateETag, callback) => {
+        // move object from overflow bucket into the main bucket
+        // retrieve initial metadata then compose the object
+        const copySource = res ||
+            createMpuKey(params.Key, params.UploadId, 'final');
+        return async.waterfall([
+            next => {
+                // retrieve metadata from init object in mpu bucket
+                const headParams = {
+                    Bucket: params.MPU,
+                    Key: createMpuKey(params.Key, params.UploadId,
+                        'init'),
+                };
+                logger.trace('retrieving object metadata');
+                return this.headObject(headParams, (err, res) => {
+                    if (err) {
+                        logHelper(logger, 'error',
+                            'err in createMultipartUpload - headObject',
+                            err);
+                        return next(err);
+                    }
+                    return next(null, res.Metadata);
+                });
+            },
+            (metadata, next) => {
+                // copy the final object into the main bucket
+                const copyMetadata = Object.assign({}, metadata);
+                copyMetadata['scal-ETag'] = aggregateETag;
+                const copyParams = {
+                    Bucket: params.Bucket,
+                    Key: params.Key,
+                    Metadata: copyMetadata,
+                    MetadataDirective: 'REPLACE',
+                    CopySource: `${params.Overflow}/${copySource}`,
+                };
+                logger.trace('copyParams', { copyParams });
+                _retryCopy.call(this, copyParams, 0, (err, res) => {
```

Comment:
There may be a good reason to do this, but why not making these functions like `_copyToMain` or `_retryCopy` member functions of a real class? (i.e declared with the `class` keyword) - you may use an actual helper class that handles MPU processing bits.